apps/wallet/src/ui/app/components/HideShowDisplayBox.tsx

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Copy16, EyeClose16, EyeOpen16 } from '@mysten/icons';
+import { useState } from 'react';
+
+import { useCopyToClipboard } from '../hooks/useCopyToClipboard';
+import { Button } from '../shared/ButtonUI';
+import { Link } from '../shared/Link';
+import { Text } from '../shared/text';
+
+export type HideShowDisplayBoxProps = {
+    value: string;
+    copiedMessage?: string;
+};
+
+export function HideShowDisplayBox({
+    value,
+    copiedMessage,
+}: HideShowDisplayBoxProps) {
+    const [valueHidden, setValueHidden] = useState(true);
+    const copyCallback = useCopyToClipboard(value, {
```

Comment:
I thought we did not allow `CopyToClipboard` because of security concerns.

Reply:
That's a good point. Although the only way to practically use it is copy maybe we should hide the copy functionality. What do you think @mystie711

Reply:
Slack thread conclusion for reference
https://mysten-labs.slack.com/archives/C042YF60NH2/p1677156381118669